### PR TITLE
Adding S3ArchiveSync construct

### PR DESCRIPTION
**Pull request by stephenwiebe**
This adds a construct that can be used to sync data stored in an S3 zip file to an EFS filesystem.

### CODE REVIEW

**Comment by pahud** [comment]
Thank you for your contribution. Will look into it shortly.

File: src/s3-archive-sync.ts [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  readonly zipFilePath: string;
+
+  /**
+   * If this is set to true, then whenever a new object is uploaded to the specified path, an EFS sync will be triggered.
+   * Currently, this functionality depends on at least on CloudTrail Trail existing in your account that captures the S3
+   * event.
+   *
+   * (optional, default: true)
+   */
+  readonly shouldMonitorS3Path?: boolean;
```
Comment (pahud) [inline src/s3-archive-sync.ts:29]:
```suggestion
  readonly syncOnUpdate?: boolean;
```

How about `syncOnUpdate`, which is more straigit-forward?

**Review by pahud — CHANGES_REQUESTED** [review]
LGTM. Just one minor suggestion.

File: src/s3-archive-sync.ts (hunk 1 above)
Comment (stephenwiebe) [reply src/s3-archive-sync.ts]: That makes sense, I went ahead and updated that.

**Comment by pahud** [comment]
@stephenwiebe we're almost there. Can you fix the conflicts ? thanks.

**Comment by stephenwiebe** [comment]
> @stephenwiebe we're almost there. Can you fix the conflicts ? thanks.

Should be good now, I think.

**Comment by pahud** [comment]
> edit: Looks like the build failed due to a tamper check... I'm not quite sure what the problem is...

It's due to the `yarn upgrade` I think.

Can you run `npx projen` again? It should fix the newline locally and you can commit again.

this will fix the anti-tamper error.

**Comment by stephenwiebe** [comment]
> Can you run `npx projen` again? It should fix the newline locally and you can commit again.
> 
> this will fix the anti-tamper error.

Ok, thanks, this looks to be fixed now.

File: README.md [hunk 2]
```diff
@@ -51,6 +50,59 @@ new GithubSourceSync(stack, 'GithubSourceSync', {
...
+const bucket = Bucket.fromBucketName(this, 'Bucket', 'demo-bucket');
+
+// Will sync initial data from compressed S3 archive to EFS, and resync if the zip file in S3 changes
+new S3ArchiveSync(this, 'S3ArchiveSync', {
+  bucket: bucket,
+  zipFilePath: 'folder/foo.zip',
+  vpc: vpc,
+  efsAccessPoint: efsAccessPoint,
+  runsAfter: [fs.mountTargetsAvailable],
+});
```
Comment (pahud) [inline README.md:104]:
```suggestion
new S3ArchiveSync(this, 'S3ArchiveSync', {
  bucket,
  zipFilePath: 'folder/foo.zip',
  vpc,
  efsAccessPoint,
  runsAfter: [fs.mountTargetsAvailable],
});
```

Let's make it looks cleaner this way, what do you think?

**Review by pahud — CHANGES_REQUESTED** [review]
we're almost there. One last minor sugestion.

**Review by pahud — APPROVED** [review]

**Comment by pahud** [comment]
Hi @stephenwiebe 

Can you run `yarn upgrade && npx projen && yarn bulid` and commit everything updated again?

I think this will help us pass the anti-tamper check.

**Comment by pahud** [comment]
Thank you so much! I will create a tweet to celebrate this!

@stephenwiebe

**Comment by stephenwiebe** [comment]
Sweet! Thanks for your help getting it merged in.

**Pull request merged**